### ISSUE TITLE
Dropping bin/* files for development in the gem?

### ISSUE THREAD

**Issue by junaruga** [ruby/rbs#673]
It's similar with https://github.com/ruby/power_assert/issues/35 and https://github.com/ruby/rake/issues/385 .

I find there are some files such as `run_in_md.rb`, `test_runner.rb`, `console` and `setup` included under the `bin/` directory in the rbs gem file. And these files might be only used for the development. Is that right? What do you think about not including the files for development in the rbs gem file?

```
$ gem fetch rbs
Fetching rbs-1.2.0.gem
Downloaded rbs-1.2.0

$ gem unpack rbs-1.2.0.gem

$ ls rbs-1.2.0/bin/
annotate-with-rdoc*  query-rdoc*  run_in_md.rb  sort*   test_runner.rb*
console*             rbs-prof*    setup*        steep*
```

Seeing the current `rbs.gemspec` excluding specific files by `Array#reject`, perhaps a way to include only necessary files might be a way not to include the bin/* files.
ex. https://github.com/puma/puma/blob/e870ab69f03e44a80c60df8b3ac1b42b1374f907/puma.gemspec#L16-L17

I found this issue when running a lint program for Ruby's RPM package in the Fedora project. I think the messages `E: non-executable-script` themselves are not related to this `rbs` project. It might be an issue related to Ruby RPM or ruby/ruby.

```
rubygem-rbs.noarch: E: non-executable-script /usr/share/gems/gems/rbs-1.0.0/bin/annotate-with-rdoc 644 /usr/bin/env ruby
rubygem-rbs.noarch: E: non-executable-script /usr/share/gems/gems/rbs-1.0.0/bin/console 644 /usr/bin/env ruby
rubygem-rbs.noarch: E: non-executable-script /usr/share/gems/gems/rbs-1.0.0/bin/query-rdoc 644 /usr/bin/env ruby
rubygem-rbs.noarch: E: non-executable-script /usr/share/gems/gems/rbs-1.0.0/bin/rbs-prof 644 /usr/bin/env ruby
rubygem-rbs.noarch: E: non-executable-script /usr/share/gems/gems/rbs-1.0.0/bin/setup 644 /usr/bin/env bash
rubygem-rbs.noarch: E: non-executable-script /usr/share/gems/gems/rbs-1.0.0/bin/sort 644 /usr/bin/env ruby
rubygem-rbs.noarch: E: non-executable-script /usr/share/gems/gems/rbs-1.0.0/bin/steep 644 /bin/sh
rubygem-rbs.noarch: E: non-executable-script /usr/share/gems/gems/rbs-1.0.0/bin/test_runner.rb 644 /usr/bin/env ruby
```

Thanks!

**Comment by pocke** [comment]
Thank you for reporting the issue!


> And these files might be only used for the development. Is that right?

It's true. These scripts are necessary only for developing RBS.
So I think removing them from the released package is a good idea. 


But I'm not sure the lint is appropriate. The files under `bin/` are not marked as "executables". https://github.com/ruby/rbs/blob/a2ce09d590f687794777daaa0451b109d63b86f6/rbs.gemspec#L35 
The files are unnecessary for end-users, but they don't affect to the end-users. So I think the lint rule is a bit misleading. 



By the way, perhaps `bundle gem`'s boilerplate should remove `bin/` directory from released files by default.

**Comment by junaruga** [comment]
Thank you for considering this!

> But I'm not sure the lint is appropriate. The files under bin/ are not marked as "executables". 

The lint program was checking the `rbs` installed as a bundled gem by Ruby's `make install` not the `rbs` installed by `gem install`. We found the Ruby's `make install` overrides the bundled gem's file permissions with 644 removing executable bits. Here is the ticket I opened: https://bugs.ruby-lang.org/issues/17840 .
